### PR TITLE
[improve][broker] Expose Netty threads num config to bookkeeper.conf

### PR DESCRIPTION
**Pull request by hangc0276**
### Motivation
https://github.com/apache/bookkeeper/pull/1612 this PR make the Netty io threads number configurable, but doesn't expose to the `bookkeeper.conf`. This parameter will be usually changed in performance tuning especially deploy this bookie on bare metal which have many cpus. For operation engineer who is unfamiliar with the source code will be hard to do performance tuning.

### Modification
Expose this parameter configuration into `bookkeeper.conf` and add more explanation about it. 

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by mattisonchao — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

File: conf/bookkeeper.conf [hunk 1]
```diff
@@ -158,6 +158,10 @@ numReadWorkerThreads=8
 # (i.e. recovery reads and adds, and fencing).
 numHighPriorityWorkerThreads=8
 
+# This is the number of threads used by Netty to handle TCP connections.
+# Default is 2 *  Runtime.getRuntime().availableProcessors()
+# serverNumIOThreads=
```
Comment (merlimat) [inline conf/bookkeeper.conf:163]: We should leave uncommented, so it will be easier to override in K8S with just env variables.

**Review by eolivelli — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@momo-jun a soft reminder: here is a PR targeted at 2.11 w/ doc-required label, could u pls follow up? Thanks

**Comment by momo-jun** [comment]
Hi @hangc0276 will you commit the doc changes (by updating the `reference-configuration.md` file?) in this PR or in a follow-up doc PR? Just let me know your plan or anything I can help with.

File: conf/bookkeeper.conf [hunk 2]
```diff
@@ -158,6 +158,10 @@ numReadWorkerThreads=8
 # (i.e. recovery reads and adds, and fencing).
 numHighPriorityWorkerThreads=8
 
+# This is the number of threads used by Netty to handle TCP connections.
+# Default is 2 *  Runtime.getRuntime().availableProcessors()
+serverNumIOThreads=
```
Comment (momo-jun) [inline conf/bookkeeper.conf:163]: The parameter name (`serverNumIOThreads`) is different from that of broker (`numIOThreads`). Does it make sense to keep it aligned?

**Review by codelipenghui — APPROVED** [review]

File: conf/bookkeeper.conf (hunk 2 above)
Comment (hangc0276) [reply conf/bookkeeper.conf:163]: They are different parameters

**Review by gaoran10 — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

**Comment by congbobo184** [comment]
@hangc0276   hi, I move this PR to release/2.9.5, if you have any questions, please ping me. thanks.

**Review by tisonkun — APPROVED** [review]
+1

Rebase and trigger the CI tasks.

**Comment by tisonkun** [comment]
CI steadily fails. Closed.

You can resubmit this patch and try to fix the tests.

**Pull request closed without merging**